activation/nipst.go

```diff
@@ -72,40 +74,68 @@ type builderState struct {
 	PoetProofRef []byte
 }
 
-func (s *builderState) load() {
+func nipstBuildStateKey() []byte {
+	return []byte("nipstate")
+}
 
+func (s *NIPSTBuilder) load(challenge types.Hash32) {
+	bts, err := s.store.Get(nipstBuildStateKey())
+	if err != nil {
+		s.log.Warning("cannot load Nipst state %v", err)
+		return
+	}
+	if len(bts) > 0 {
+		var state builderState
+		err = types.BytesToInterface(bts, &state)
+		if err != nil {
+			s.log.Error("cannot load Nipst state %v", err)
+		}
+		if state.Challenge == challenge {
+			s.state = &state
+		} else {
+			s.state = &builderState{Challenge: challenge, Nipst: &types.NIPST{}}
+		}
+	}
 }
 
-func (s *builderState) persist() {
-	// TODO(noamnelke): implement
+func (s *NIPSTBuilder) persist() {
+	bts, err := types.InterfaceToBytes(&s.state)
+	if err != nil {
+		s.log.Warning("cannot store Nipst state %v", err)
+		return
+	}
+	err = s.store.Put(nipstBuildStateKey(), bts)
+	if err != nil {
+		s.log.Warning("cannot store Nipst state %v", err)
+	}
 }
 
 type NIPSTBuilder struct {
 	id         []byte
 	postProver PostProverClient
 	poetProver PoetProvingServiceClient
-	poetDb     PoetDb
+	poetDb     PoetDbApi
```

Comment:
I'd do - PoetDBAPI, even though it's a lot of upper case stuff. it's the right way